pkg/handlers/generic/validation/helm/handler.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package helm
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/api/variables"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pkg/handlers/generic/clusterconfig"
+)
+
+type HelmRegistryValidator struct {
+	client       ctrlclient.Client
+	variableName string
+}
+
+func New(
+	c ctrlclient.Client,
+) *HelmRegistryValidator {
+	return &HelmRegistryValidator{
+		client:       c,
+		variableName: clusterconfig.MetaVariableName,
+	}
+}
+
+func (h *HelmRegistryValidator) Name() string {
+	return "HelmRegistryValidator"
+}
+
+func (h *HelmRegistryValidator) ValidateTopology(
```

Comment:
can we please add a unit test for this

Reply:
Im not sure about this particular function, but we can split it out to be more testable